bb8/src/inner.rs

```diff
@@ -149,12 +149,19 @@ where
             "handled in caller"
         );
 
+        let max_lifetime = self.inner.statics.max_lifetime;
+        let is_expired = max_lifetime.map_or(false, |lt| conn.is_expired(Instant::now() - lt));
```

Comment:
Suggest we bind `let is_broken = self.inner.manager.has_broken(&mut conn.conn);` here and then use `is_broken || is_expired` in the match scrutinee.